File: sleep_front/src/types/enums/StatusPedido.ts
export const StatusPedido = {
  PENDENTE: "PENDENTE",
  APROVADO: "APROVADO",
  CANCELADO: "CANCELADO",
} as const;

export type StatusPedido = typeof StatusPedido[keyof typeof StatusPedido];

export const StatusPedidoLabel: Record<StatusPedido, string> = {
  PENDENTE: "Pedndente",
  APROVADO: "Aprovado",
  CANCELADO: "Cancelado",
};

File: src/main/java/sleep/colchoes/domain/enums/StatusPedido.java
package sleep.colchoes.domain.enums;

public enum StatusPedido {
    PENDENTE("P"),
    APROVADO("A"),
    CANCELADO("C");

    private final String codigo;

    StatusPedido(String codigo) {
        this.codigo = codigo;
    }

    public String getCodigo() {
        return codigo;
    }

    public static StatusPedido fromCodigo(String codigo) {
        if (codigo == null) return null;
        return switch (codigo.toUpperCase()) {
            case "P" -> PENDENTE;
            case "A" -> APROVADO;
            case "C" -> CANCELADO;
            default -> throw new IllegalArgumentException("Código inválido: " + codigo);
        };

    }
}